nifi-nar-bundles/nifi-standard-bundle/nifi-standard-parameter-providers/src/main/java/org/apache/nifi/parameter/FileParameterProvider.java

```diff
@@ -0,0 +1,242 @@
+                    .explanation(String.format("Fetched %s files as parameters.", parameterCount))
+                    .build());
+        } catch (final IllegalArgumentException e) {
+            verificationLogger.error("Failed to fetch parameters", e);
+            results.add(new ConfigVerificationResult.Builder()
+                    .outcome(ConfigVerificationResult.Outcome.FAILED)
+                    .verificationStepName("Fetch Parameters")
+                    .explanation("Failed to fetch parameters: " + e.getMessage())
+                    .build());
+        }
+        return results;
+    }
+
+    private String getParameterValue(final String rawValue, final ParameterValueEncoding encoding) {
+        if (ParameterValueEncoding.BASE64 == encoding) {
+            return new String(Base64.getDecoder().decode(rawValue.getBytes(DEFAULT_CHARSET)), DEFAULT_CHARSET);
+        }
+        return rawValue;
+    }
+
+    private Collection<File> getDirectories(final ConfigurationContext context, final PropertyDescriptor descriptor) {
+        return context.getProperty(descriptor).isSet()
+                ? Arrays.stream(context.getProperty(descriptor).getValue().split(",")).map(String::trim).map(File::new).collect(Collectors.toSet())
+                : Collections.emptySet();
+    }
+
+    private ParameterGroup getParameterGroup(final ConfigurationContext context, final File directory, final String groupName) {
+        final int parameterSizeLimit = context.getProperty(PARAMETER_VALUE_BYTE_LIMIT).asDataSize(DataUnit.B).intValue();
+        final ParameterValueEncoding parameterEncoding = ParameterValueEncoding.valueOf(context.getProperty(PARAMETER_VALUE_ENCODING).getValue().toUpperCase());
+
+        final File[] files = directory.listFiles();
+
+        final List<Parameter> parameters = new ArrayList<>();
+        for (final File file : files) {
+            if (file.isDirectory() || file.isHidden()) {
+                continue;
+            }
+
+            final String parameterName = file.getName();
+
+            try (final InputStream in = new BufferedInputStream(new LimitingInputStream(new FileInputStream(file), parameterSizeLimit))) {
+                final String rawValue = IOUtils.toString(in, Charset.defaultCharset()).trim();
+                final String parameterValue = getParameterValue(rawValue, parameterEncoding);
+
+                if (parameterValue.length() >= parameterSizeLimit) {
+                    getLogger().warn("Parameter {} may be truncated at {} bytes", parameterName, parameterValue.length());
+                }
+
+                final ParameterDescriptor parameterDescriptor = new ParameterDescriptor.Builder().name(parameterName).build();
+                parameters.add(new Parameter(parameterDescriptor, parameterValue, null, true));
+            } catch (final IOException e) {
+                getLogger().error("Failed to read file [{}]", new Object[] { file }, e);
```

Comment:
Is it necessary to log an error and throw an exception, or can this log be removed?
```suggestion
                getLogger().error("Failed to read file [{}]", file, e);
```